src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java

```diff
@@ -139,26 +156,29 @@ public static ExternalTool parseAddExternalToolManifest(String manifest) {
         
         // Allow IllegalArgumentException to bubble up from ExternalTool.Type.fromString
         ExternalTool.Type type = ExternalTool.Type.fromString(typeUserInput);
+        ExternalTool.Scope scope = ExternalTool.Scope.fromString(scopeUserInput);
         String toolUrl = getRequiredTopLevelField(jsonObject, TOOL_URL);
         JsonObject toolParametersObj = jsonObject.getJsonObject(TOOL_PARAMETERS);
         JsonArray queryParams = toolParametersObj.getJsonArray("queryParameters");
         boolean allRequiredReservedWordsFound = false;
-        for (JsonObject queryParam : queryParams.getValuesAs(JsonObject.class)) {
-            Set<String> keyValuePair = queryParam.keySet();
-            for (String key : keyValuePair) {
-                String value = queryParam.getString(key);
-                ReservedWord reservedWord = ReservedWord.fromString(value);
-                if (reservedWord.equals(ReservedWord.FILE_ID)) {
-                    allRequiredReservedWordsFound = true;
+        if (scope.equals(Scope.FILE)) {
```

Comment:
else Scope.DATASET - just not written yet?

Reply:
Well, I'm not sure. Should dataset tools have any reserved words that are required?

Reply:
without an else statement, allRequiredReservedWordsFound is false and an exception happens so this method doesn't work for scope dataset as is. 
w.r.t. reserved words - it would be odd/bad to configure a dataset tool to not get the dataset id (or pid) in the same way that a file tool needs a file id (or pid if that's implemented).

Reply:
ah sorry - I see the exception is in the if statement - probably should not define the allRequiredWordsFound outside the if clause then (to avoid confusing people like me :-) )